contracts/protocol/facets/MetaTransactionsHandlerFacet.sol

```diff
@@ -305,4 +316,41 @@ contract MetaTransactionsHandlerFacet is IBosonMetaTransactionsHandler, Protocol
 
         return executeTx(_userAddress, _functionName, _functionSignature, _nonce);
     }
+
+    /**
+     * @notice Manages allow list of functions that can be executed using metatransactions.
+     *
+     * Emits a FunctionsWhitelisted event if successful.
+     *
+     * Reverts if:
+     * - Caller is not a protocol admin
+     *
+     * @param _functionNameHashes - a list of hashed function names (keccak256)
+     * @param _isWhitelisted - new whitelist status
+     */
+    function setWhitelistedFunctions(bytes32[] calldata _functionNameHashes, bool _isWhitelisted)
+        public
+        override
+        onlyRole(ADMIN)
+    {
+        ProtocolLib.ProtocolMetaTxInfo storage pmti = protocolMetaTxInfo();
+
+        // set new values
+        for (uint256 i = 0; i < _functionNameHashes.length; i++) {
+            pmti.isWhitelisted[_functionNameHashes[i]] = _isWhitelisted;
+        }
+
+        // Notify external observers
+        emit FunctionsWhitelisted(_functionNameHashes, _isWhitelisted, msgSender());
```

Comment:
I agree strings aren't efficient to store, but if we only emit the keccak256 hash of the function, we'll never really be able to see which functions are white listed. It might be better to pass in and emit an array of strings but  hash and store them in the function as bytes32 hash.

Reply:
Changing `isFunctionWhitelisted` to take a string could be a good solution.Otherwise, I'm afraid we're going to be completely blind. Let's discuss with the team.